gpytorch/kernels/constant_kernel.py

```diff
@@ -0,0 +1,119 @@
+from ..constraints import Interval, Positive
+from ..priors import Prior
+from .kernel import Kernel
+
+
+class ConstantKernel(Kernel):
+    """
+    Constant covariance kernel for the probabilistic inference of constant coefficients.
+
+    ConstantKernel represents the prior variance `k(x1, x2) = var(c)` of a constant `c`.
+    The prior variance of the constant is optimized during the GP hyper-parameter
+    optimization stage. The actual value of the constant is computed (implicitly) using
+    the linear algebraic approaches for the computation of GP samples and posteriors.
+
+    The kernel (`k_constant`) is most useful as a modification of an arbitrary `k_base`:
+        1) Additive constants: The modification `k_base + k_constant` allows the GP to
+            infer a non-zero asymptotic value far from the training data, which
+            generally leads to more accurate extrapolation. Notably, the uncertainty in
+            this constant value affects the posterior covariances through the posterior
+            inference equations. This is not the case when a constant prior mean is
+            used, since the prior mean does not show up the posterior covariance and is
+            not regularized by the log-determinant during the optimization of the marginal
+            likelihood.
+        2) Multiplicative constants: The modification `k_base * k_constant` allows the
+            GP to modulate the variance of the kernel `k_base`, and is mathematically
+            identical to `ScaleKernel(base_kernel)` with the same constant.
+    """
+
+    has_lengthscale = False
+
+    def __init__(
+        self,
+        batch_shape: Optional[torch.Size] = None,
+        constant_prior: Optional[Prior] = None,
+        constant_constraint: Optional[Interval] = None,
+        active_dims: Optional[Tuple[int, ...]] = None,
+    ):
+        """Constructor of ConstantKernel.
+
+        Args:
+            batch_shape: The batch shape of the kernel.
+            constant_prior: Prior over the constant parameter.
+            constant_constraint: Constraint to place on constant parameter.
+        """
+        super().__init__(batch_shape=batch_shape, active_dims=active_dims)
+
+        self.register_parameter(
+            name="raw_constant",
+            parameter=torch.nn.Parameter(torch.zeros(*self.batch_shape, 1)),
+        )
+
+        if constant_prior is not None:
+            if not isinstance(constant_prior, Prior):
+                raise TypeError("Expected gpytorch.priors.Prior but got " + type(constant_prior).__name__)
+            self.register_prior(
+                "constant_prior",
+                constant_prior,
+                lambda m: m.constant,
+                lambda m, v: m._set_constant(v),
+            )
+
+        if constant_constraint is None:
+            constant_constraint = Positive()
+        self.register_constraint("raw_constant", constant_constraint)
+
+    @property
+    def constant(self) -> torch.Tensor:
```

Comment:
nit: import `Tensor` to make the type hints less verbose
```suggestion
    def constant(self) -> Tensor:
```